spacy/pipeline/pipes.pyx

```diff
@@ -1033,8 +1033,7 @@ class TextCategorizer(Pipe):
         return 1
 
     def begin_training(self, get_gold_tuples=lambda: [], pipeline=None, sgd=None, **kwargs):
-        for raw_text, annots_brackets in get_gold_tuples():
-            cats = annots_brackets.pop()
+        for raw_text, (_, (cats, _2)) in get_gold_tuples():
             for cat in cats:
                 self.add_label(cat)
```

Comment:
This doesn't work because the cats-brackets are on each sentence. It needs to be something more like:

        for raw_text, annots_brackets in get_gold_tuples():
            for annots, (cats, brackets) in annots_brackets:
                for cat in cats:
                    self.add_label(cat)

Or possibly the problem is that `merge_sents` isn't called for some reason in some cases, but in my textcat test cases this doesn't currently work.